src/v/kafka/server/group_manager.cc

```diff
@@ -85,6 +85,9 @@ ss::future<> group_manager::stop() {
     _pm.local().unregister_manage_notification(_manage_notify_handle);
     _gm.local().unregister_leadership_notification(_leader_notify_handle);
 
+    // cancel any pending operations from upcalls
+    _lock.broken();
```

Comment:
It is hard to understand the change from y perspective. Why do we need to prevent concurrent topic deletion gc. 

To me, concurrency would happen with the `ss:;gate()` so why is the mutex needed. can you comment on that.